cmd/dep/status.go

```diff
@@ -1071,6 +1084,32 @@ func collectConstraints(ctx *dep.Ctx, p *dep.Project, sm gps.SourceManager) (con
 	return constraintCollection, errs
 }
 
+// Dep status iterates in several places over a list of projects and executes
+// code per project in parallel. These goroutines can create additional file
+// descriptors for example by writing to /dev/null via ioutil.Discard or by
+// creating pipes to communicate with subprocesses (e.g. executing git
+// via cmd/vcs_source). If the project list is long enough, it can result in
+// hitting the file descriptor limit set in the operating system.
+//
+// getConcurrentProjectLimit provides a limit how many goroutines should be
+// run in parallel. It attempts to read the soft RLIMIT_NOFILE from the OS.
+// If this fails, it takes the default limit of 256 which is the current default
+// limit on Darwin ( and lower than Linux/Win ). This limit is divided
+// by how many FDs are used max in parallel per goroutine (A first
+// approximation through profiling the execution of dep status is 4.)
+// TODO: Improve max parallel FDs per goroutine approximation.
+func getConcurrentProjectLimit(ctx *dep.Ctx) uint64 {
+	var maxFDperGoroutine uint64 = 4
+	var defaultSoftRlimit uint64 = 256
+	var rLimit syscall.Rlimit
+	err := syscall.Getrlimit(syscall.RLIMIT_NOFILE, &rLimit)
```

Comment:
I think this would work only on *nix systems. We should have separate functions for other platforms.

> Syscall must always be guarded with build tags.
                - go-proverbs

Reply:
Agreed - if we can make this a bit more cross-platform safe, i think it's a good addition.